src/Textarea.jsx

```diff
@@ -0,0 +1,107 @@
+import React from 'react';
+import cx from 'classnames';
+import autosize from 'autosize';
+
+/**
+ * @module Textarea
+ */
+class Textarea extends React.Component {
+	constructor(props) {
+		super(props);
+		this.state = {
+			value: props.value || '',
+		};
+		this.onChange = this.onChange.bind(this);
+	}
+
+	onChange(e) {
+		this.setState({
+			value: e.target.value,
+		});
+	}
+
+	componentDidMount() {
+		if (this.props.autoheight){
+			autosize(this.textarea);
```

Comment:
the only thing I'm wondering about this is if the 'autosize' value might sometimes get lost when the element re-renders, e.g. when navigating away from the page and hitting 'back'. Worth doing some experimenting about whether you need to re-initialize it on every `render` but hopefully this `componentDidMount` is sufficient